pkg/snapshot/snapshotter/snapshotter.go

```diff
@@ -740,3 +744,21 @@ func (ssr *Snapshotter) checkSnapstoreSecretUpdate() bool {
 	snapstoreHash[ssr.snapstoreConfig.Provider] = newSnapstoreSecretHash
 	return true
 }
+
+// IsScheduledFullSnapshotMissed checked whether the last scheduled full-snapshot is missed or scheduled full-snapshot can be missed.
+func (ssr *Snapshotter) IsScheduledFullSnapshotMissed() bool {
```

Comment:
Please segregate this method into two methods, one to check whether previous scheduled full snapshot was missed, and one to check whether the upcoming scheduled full snapshot might be missed.

```suggestion
// WasScheduledFullSnapshotMissed checks whether the last scheduled full-snapshot was missed or scheduled full-snapshot can be missed.
func (ssr *Snapshotter) WasScheduledFullSnapshotMissed() bool {
```

Reply:
Yes or rename this method as this method now does not do what the name suggests